src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -297,6 +298,10 @@ internal static LoggerFilterOptions CreateFilterOptions(LoggerFilterOptions regi
             QuickPulseInitializationScheduler delayer,
             Action<TelemetryConfiguration> additionalTelemetryConfig)
         {
+            if (options.DiagnosticsEventListenerLogLevel != null)
+            {
+                _ = new ApplicationInsightsEventListener((EventLevel)options.DiagnosticsEventListenerLogLevel);
```

Comment:
I think this might be optimized away or GC'd - there is no persistent reference to it. Look into using a `ITelemetryModule`, which is an AppInsights SDK hook to supply a generate piece of code to start up with the appinsights SDK

Reply:
Added TelemetryModule